tests/unit/containers/my-cooperations/cooperation-activities-list/CooperationActivitiesList.spec.jsx

```diff
@@ -1,43 +1,224 @@
+    setCurrentSectionIndex: vi.fn()
+  })
+
+  renderWithProviders(
+    <CooperationActivitiesList
+      data={mockedData}
+      handleNonInputValueChange={handleNonInputValueChange}
+    />
+  )
+}
+
+describe('CooperationActivitiesList with section data', () => {
+  const mockedSectionsData = {
+    sections: [
+      {
+        title: 'Section1 title',
+        description: 'Section1 description',
+        order: ['66183816fb40f35f91bb77ce'],
+        lessons: [
+          {
+            _id: '66183816fb40f35f91bb77ce',
+            title: 'Lesson 1',
+            description: 'Lesson 1 description',
+            content: 'Lesson 1 content',
+            resourceType: 'lessons'
+          }
+        ],
+        quizzes: [],
+        attachments: [],
+        id: '17121748017180'
+      },
+      {
+        title: 'Section2 title',
+        description: 'Section2 description',
+        lessons: [],
+        quizzes: [],
+        attachments: [],
+        id: '17121748017181'
+      }
+    ]
+  }
 
-    fireEvent.mouseOver(hoverElement[0])
+  beforeEach(() => {
+    renderWithMockData(mockedSectionsData, 0)
+  })
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
 
-    const addButton = screen.getAllByText('Add activity')
+  it('should add a new section when Add activity button is clicked', async () => {
+    const [hoverElement] = await screen.findAllByTestId('addActivity-container')
+    fireEvent.mouseOver(hoverElement)
 
-    fireEvent.click(addButton[0])
-    const menuItem = await screen.findAllByText(
+    const [addButton] = screen.getAllByText('Add activity')
+    fireEvent.click(addButton)
+
+    const [menuItem] = await screen.findAllByText(
       'cooperationsPage.manyTypes.module'
     )
-    fireEvent.click(menuItem[0])
+    fireEvent.click(menuItem)
+
     const sections = await screen.findAllByTestId('addActivity-container')
     expect(sections.length).toBe(2)
   })
+
+  it('should delete section resource', async () => {
+    await waitFor(() => {
+      const deleteResourceBtn = screen.getByTestId('CloseIcon').parentElement
+      fireEvent.click(deleteResourceBtn)
+    })
+
+    const resource = screen.queryByText('Lesson 1 description')
+
+    await waitFor(() => {
+      expect(resource).not.toBeInTheDocument()
+    })
+  })
+
+  it('should change the activity title', async () => {
+    const titleInput = await screen.findByDisplayValue(
+      mockedSectionsData.sections[0].title
+    )
+    const newTitle = 'New section title'
+
+    fireEvent.blur(titleInput, {
+      target: { value: newTitle }
+    })
+
+    await waitFor(() => {
+      expect(titleInput.value).toBe(newTitle)
+    })
+  })
+
+  it('should set sections from the data first in order if the current section index is undefined', async () => {
+    renderWithMockData(mockedSectionsData, undefined)
+
+    const newSectionData = mockedCourseData.sections.map((section, index) => ({
+      ...section,
+      id: Date.now().toString() + index
```

Comment:
I have noticed that code piece `Date.now().toString()` is repeated frequently. Do we really need to mock `Date.now()` to then generate id with `Date.now().toString()`? Can't we use just hardcoded `id` instead?

Reply:
I can't use hardcoded ids here because `newSectionData` is created in the component using the following function:
```javascript
const newSectionData = selectedCourse.sections.map(
  (section, index) => ({
    ...section,
    id: Date.now().toString() + index
  })
)
```
And I need to check that the data passed to `handleNonInputValueChange` in the test is the same as in the component. To do this, I have to reproduce the function's behavior.